pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1214,4 +1273,112 @@ private int getMaxNumPartitionsPerInstance(InstancePartitions instancePartitions
       return (numPartitions + numInstancesPerReplicaGroup - 1) / numInstancesPerReplicaGroup;
     }
   }
+
+  // Pre fetch the LLC segment without deep store copy.
+  public void prefetchLLCSegmentWithoutDeepStoreCopy(String tableNameWithType) {
```

Comment:
why public?

Reply:
prefetchLLCSegmentsWithoutDeepStoreCopy

Reply:
RealtimeSegmentValidationManager [uses this method. ](https://github.com/apache/incubator-pinot/pull/6778/files/647f1de17ff0bc60bef20468604e65b44dbc057c#diff-0a9ffdab99ed314e7a8e58bf86c125f020b7d6f86493d1f6d719beab62b7f106R77)